test/proxy.t.sol

```diff
@@ -52,93 +50,229 @@ contract ProxyTest is DSTest {
-        uint b = Proxy(proxyB).accessToken();
-        assertEq(proxyB, registry.proxies(b));
+        proxy.file("target", _target);
+        assertEq(proxy.target(), _target);
     }
 
-    function testExecute() public {
+    function testExecute(address user) public {
         address payable proxyAddr = registry.build();
         Proxy proxy = Proxy(proxyAddr);
 
         bytes memory data = abi.encodeWithSignature("inlineAdd(uint256,uint256)", 5,7);
 
-        // action not calling other method
-        bytes memory response = proxy.execute(address(action), data);
+        // set action as a safe target
+        proxy.file("target", address(action));
+
+        // Add this as a user so it can execute
+        proxy.addUser(user);
+
+        // execute action that does not call core contract
+        vm.prank(user);
+        bytes memory response = proxy.userExecute(address(action), data);
 
-        // using core
+        // execute action that does call core contract
         data = abi.encodeWithSignature("doAdd(address,uint256,uint256)", address(core), 5,7);
-        response = proxy.execute(address(action), data);
+        vm.prank(user);
+        response = proxy.userExecute(address(action), data);
 
         // msg.sender should be proxy address
         assertEq(core.caller(), proxyAddr);
-
     }
 
-    function testFailExecuteAccessActionStorage() public {
+    function testFailExecuteWithBadData(address user) public {
         address payable proxyAddr = registry.build();
         Proxy proxy = Proxy(proxyAddr);
 
-        // using action contract storage should fail
-        bytes memory data = abi.encodeWithSignature("doAdd(uint256,uint256)", address(core), 5,7);
-        bytes memory response = proxy.execute(address(action), data);
+        // use non-existant function call
+        bytes memory data = abi.encodeWithSignature("inlineSubtract(uint256,uint256)", 5,7);
+
+        // set action as a safe target
+        proxy.file("target", address(action));
+
+        // Add this as a user so it can execute
+        proxy.addUser(user);
+
+        // execute action that does not call core contract
+        vm.prank(user);
+        bytes memory response = proxy.userExecute(address(action), data);
     }
 
-    function testFailExecuteNotNFTOwner() public {
+    function testExecuteNotUserFails() public {
         address payable proxyAddr = registry.build();
         Proxy proxy = Proxy(proxyAddr);
 
-        uint accessToken = proxy.accessToken();
-        registry.transferFrom(msg.sender,address(123), accessToken);
+        bytes memory data = abi.encodeWithSignature("inlineAdd(uint256,uint256)", 5,7);
+
+        // set action as a safe target
+        proxy.file("target", address(action));
 
-        // using core
-        bytes memory data = abi.encodeWithSignature("doAdd(address,uint256,uint256)", address(core), 5,7);
+        // Add this as a user so it can execute
+        // proxy.addUser(address(this));
 
```

Comment:
```suggestion
```

Reply:
Ah, these were actually intentional as a lot of the tests followed the same structure, so I'd comment sections out to show what that test was doing differently. Looks a bit sloppy though, and makes less sense now that there's a lot more tests